submissions/DmytryjK/memory-pair-game/js/script.js

```diff
@@ -0,0 +1,197 @@
+    },
+    { hero : 'lancer',
+      srcHero: 'images/images_cards/lancer.jpg'
+    },
+    { hero : 'gyro',
+      srcHero: 'images/images_cards/gyro.jpg'
+    },
+    { hero : 'necr',
+      srcHero: 'images/images_cards/necr.jpg'
+    },
+    { hero : 'void',
+      srcHero: 'images/images_cards/void.jpg'
+    }
+];
+
+const srcBg = 'images/back/dota_back.svg';
+let parrentCardBlock = document.querySelector('.main__cards');
+let winnerPopup = document.querySelector('.winMessage');
+let doubleElementsOfCards = [];
+
+createElementsOfCards();
+
+let listCards = document.querySelectorAll('.main__card'),
+    activeCards = [],
+    numbersOfCloseCards = 0;
+
+let currentTime = document.querySelector('.timer-current span'),
+    bestTime = document.querySelector('.timer-best span'),
+    minutes = 0, 
+    seconds = 0,
+    bestResultsSheet = [];
+let timerInterval;
+
+setTimeout(showAllCards, 400);
+setTimeout(hideAllCards, 2000);
+
+if (localStorage.getItem('bestTime') !== null) {
+  bestTime.innerText = localStorage.getItem('bestTime');
+  bestResultsSheet.push(bestTime.innerText);
+}
+
+parrentCardBlock.addEventListener('click', e => {
+    const parrentTarget = e.target.parentNode;
+    
+    if (parrentTarget.classList.contains('main__card')) {
+        parrentTarget.classList.add('active-card');
+        parrentTarget.style.cssText += 'pointer-events: none';
+        activeCards.push(parrentTarget);
+    }
+
+    if (activeCards.length >= 2) {
+        listCards.forEach(e => {
+            e.style.cssText += 'pointer-events: none';
+        });
+
+        if(activeCards[0].getAttribute(["data-name"]) === activeCards[1].getAttribute(["data-name"]) && activeCards[0].getAttribute(["data-numbers"]) !== activeCards[1].getAttribute(["data-numbers"])) {
+            setTimeout(hideTwoCards, 600, activeCards);
+        } else {
+            setTimeout(closeTwoCards, 800, activeCards);
+        }
+  
+        if (numbersOfCloseCards === doubleElementsOfCards.length / 2 - 1) {
+            const buttonToContinue = document.querySelector('.click-to-continue');
+
+            winnerPopup.classList.add('winMessage__active');
+
+            localStorage.setItem('time', currentTime.innerText);
+            currentTime.innerText = localStorage.getItem('time');
+            bestResultsSheet.push(localStorage.getItem('time'));
+
+            if (bestResultsSheet.length > 1) {
+              bestTime.innerText = bestResultsSheet.reduce((prevItem, currentItem) => {
+                if (currentItem < prevItem) {
+                  return currentItem;
+                } else {
+                  return prevItem;
+                }
+              });
+            } else {
+              bestTime.innerText = localStorage.getItem('time');
+            }
+
+            localStorage.setItem('bestTime', bestTime.innerText);
+            clearInterval(timerInterval);
+
+            buttonToContinue.addEventListener('click', () => {
```

Comment:
Move event listeners callbacks into independent functions, it's hard to read nested functions. Also, remember one thing one function one action/responsibility.

Reply:
fixed